cpp/src/io/parquet/reader_impl_helpers.cpp

```diff
@@ -1146,62 +1147,63 @@ aggregate_reader_metadata::apply_row_bounds_filter(
   auto const num_sources = input_row_group_indices.size();
 
   auto filtered_row_group_indices = std::vector<std::vector<size_type>>(num_sources);
-  auto row_group_row_counts       = std::vector<std::vector<size_type>>{};
+  auto row_group_row_counts       = std::vector<std::vector<size_t>>{};
   row_group_row_counts.reserve(num_sources);
-  auto row_group_row_offsets = std::vector<std::vector<size_type>>{};
+  auto row_group_row_offsets = std::vector<std::vector<size_t>>{};
   row_group_row_offsets.reserve(num_sources);
 
-  size_type current_row_index = 0;
+  size_t current_row_index = 0;
 
   // Rows to skip relative to the row offset of the first surviving row group index
   int64_t first_row_group_relative_rows_to_skip = 0;
 
   // For each data source
-  std::for_each(thrust::counting_iterator<size_t>(0),
-                thrust::counting_iterator(num_sources),
-                [&](auto const& src_idx) {
-                  auto const& file_metadata = per_file_metadata[src_idx];
-                  auto const num_row_groups = input_row_group_indices[src_idx].size();
-
-                  // Fill row group rows counts and row offsets with zeros
-                  row_group_row_counts.emplace_back(num_row_groups, size_type{0});
-                  row_group_row_offsets.emplace_back(num_row_groups, current_row_index);
+  std::for_each(
```

Comment:
Mostly whitespace changes in this `std::for_each`. Only real change is the use of `std::cmp` in `if` statements